subworkflows/local/domain_classification.nf

```diff
@@ -0,0 +1,143 @@
+/*
+* Domain classification with Tiara
+*/
+
+include { TIARA_TIARA                                                  } from '../../modules/nf-core/tiara/tiara/main'
+include { TIARA_CLASSIFY                                               } from '../../modules/local/tiara_classify'
+include { DASTOOL_FASTATOCONTIG2BIN as DASTOOL_FASTATOCONTIG2BIN_TIARA } from '../../modules/nf-core/dastool/fastatocontig2bin/main'
+include { COMBINE_TSV as TIARA_SUMMARY                                 } from '../../modules/local/combine_tsv'
+
+workflow DOMAIN_CLASSIFICATION {
```

Comment:
I wonder if the future users may request alternative domain classification tools, maybe it would be worth future-proofing and pushing most of the TIARA sections into a Tiara-speciufic sub workflow, and pull that into here with a if/else statement to allow user choice of tool (Currently only TIara)?

Reply:
I think this is a good idea - my only problem is that I'm not sure what files other classifiers might want in future. Currently the DOMAIN_CLASSIFICATION workflow takes the assembles, bins, and unbins, classifies the contigs for each assembly, and then uses the custom script to split the assembly contig classifications back into their bins. Other classifiers might work differently and require different files - if that's a problem we want to tackle if and when in future, happy to split Tiara into a separate workflow now, otherwise we could wait and see what a future tool requires?

Reply:
I would say tackle it when it comes later! CAn always add more input  channels to the subwf! :)